### PR TITLE
doc: metion for the Java 21's new switch statement in document

### PR DESCRIPTION
**Pull request by duobei**
#979

### CODE REVIEW

File: docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 1]
```diff
@@ -170,6 +170,40 @@ private void receiveMsg3(Msg3 msg) {
...
+
+    public static class Msg3 {}
+
+    @Override
+    public void onReceive(Object msg) throws Exception {
+      switch(msg) {
+        case Msg1 msg -> receiveMsg1((Msg1) msg);
+        case Msg2 msg -> receiveMsg2((Msg2) msg);
+        case Msg3 msg -> receiveMsg3((Msg3) msg);
+        default _ -> unhandled(msg);
```
Comment (He-Pin) [inline docs/src/test/java/jdocs/actor/ActorDocTest.java:189]: Very cool and very clean, thank you.

**Review by He-Pin — APPROVED** [review]
LGTM, thanks.

File: docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 2]
```diff
@@ -170,6 +170,40 @@ private void receiveMsg3(Msg3 msg) {
   }
   // #optimized
 
+  public
+  // #pattern-match-java21
+  static class PatternMatchedActor extends UntypedAbstractActor {
```
Comment (He-Pin) [inline docs/src/test/java/jdocs/actor/ActorDocTest.java:175]: Seems like we need a `jdk-21` support, eg `java-jdk-21`. So seems we need to update the Jdk plugin then.

File: docs/src/test/java/jdocs/actor/ActorDocTest.java (hunk 1 above)
Comment (He-Pin) [reply docs/src/test/java/jdocs/actor/ActorDocTest.java:189]: I think you should move this fild to `java-jdk-9`, thanks.

**Review by He-Pin — COMMENTED** [review]
Maybe we just attach the code directly in the doc, then there will be no error.

**Comment by duobei** [comment]
> Maybe we just attach the code directly in the doc, then there will be no error.

I'll try later.

**Review by He-Pin — APPROVED** [review]
Cool, thanks

**Comment by pjfanning** [comment]
I'm not sure about the positioning of this doc. The Classic Actors page is basically regarded as documented an effectively deprecated usage pattern. This change also goes against the norm in this .md file which allow Scala users to just see Scala code snippets. I may be wrong about how this displays and will check the docs again after the nightly build runs.

I think we would be better off having a dedicated Java 21 page and having links to that from the Classic and the Typed Actors pages. Maybe also useful to link to this from the Stream pages.

wdyt @mdedetrich

**Comment by He-Pin** [comment]
I think this is fine. And the classical actor will not go anywhere, which is very high performance.

For the doc , we can always improve it later, once we get the doc published with jdk 21, we can even let it compile checked.

I can do the split tomorrow, but as this is his first contribution, we should be more nice and welcome.

**Comment by He-Pin** [comment]
And even now, the scala user can refs to javadsl packages , that should not be a big problem, paradox support tagging, I cab improve that.

**Comment by He-Pin** [comment]
@pjfanning @mdedetrich I just chatted with the origin contributor, I will take care it now, I just opened up my laptop and it's 1.49 AM

**Comment by pjfanning** [comment]
@He-Pin there is no hurry to remedy this. In fact, I think we need to slow things down. I think PRs should be open for a few days before they are merged. I have sent an email to dev list with this proposal.

**Comment by He-Pin** [comment]
@pjfanning I have to sleep now, I think it's fine now, new contributors may not know everything about the toolchain, I think that's ok.

**Pull request merged**